worlds/cccharles/Regions.py

```diff
@@ -0,0 +1,290 @@
+
+    towers_region = Region("Towers", player, world)
+    towers_region.add_locations(loc_towers, CCCharlesLocation)
+    world.regions.append(towers_region)
+
+    north_beach_region = Region("North beach", player, world)
+    north_beach_region.add_locations(loc_north_beach, CCCharlesLocation)
+    world.regions.append(north_beach_region)
+
+    mine_shaft_region = Region("Mine Shaft", player, world)
+    mine_shaft_region.add_locations(loc_mine_shaft, CCCharlesLocation)
+    world.regions.append(mine_shaft_region)
+
+    mob_camp_region = Region("Mob Camp", player, world)
+    mob_camp_region.add_locations(loc_mob_camp, CCCharlesLocation)
+    world.regions.append(mob_camp_region)
+
+    mob_camp_locked_room_region = Region("Mob Camp Locked Room", player, world)
+    mob_camp_locked_room_region.add_locations(loc_mob_camp_locked_room, CCCharlesLocation)
+    world.regions.append(mob_camp_locked_room_region)
+
+    mine_elevator_exit_region = Region("Mine Elevator Exit", player, world)
+    mine_elevator_exit_region.add_locations(loc_mine_elevator_exit, CCCharlesLocation)
+    world.regions.append(mine_elevator_exit_region)
+
+    mountain_ruin_outside_region = Region("Mountain Ruin Outside", player, world)
+    mountain_ruin_outside_region.add_locations(loc_mountain_ruin_outside, CCCharlesLocation)
+    world.regions.append(mountain_ruin_outside_region)
+
+    mountain_ruin_inside_region = Region("Mountain Ruin Inside", player, world)
+    mountain_ruin_inside_region.add_locations(loc_mountain_ruin_inside, CCCharlesLocation)
+    world.regions.append(mountain_ruin_inside_region)
+
+    prism_temple_region = Region("Prism Temple", player, world)
+    prism_temple_region.add_locations(loc_prism_temple, CCCharlesLocation)
+    world.regions.append(prism_temple_region)
+
+    pickle_val_region = Region("Pickle Val", player, world)
+    pickle_val_region.add_locations(loc_pickle_val, CCCharlesLocation)
+    world.regions.append(pickle_val_region)
+
+    shrine_near_temple_region = Region("Shrine Near Temple", player, world)
+    shrine_near_temple_region.add_locations(loc_shrine_near_temple, CCCharlesLocation)
+    world.regions.append(shrine_near_temple_region)
+
+    morse_bunker_region = Region("Morse Bunker", player, world)
+    morse_bunker_region.add_locations(loc_morse_bunker, CCCharlesLocation)
+    world.regions.append(morse_bunker_region)
+
+    # Place "Victory" event at "Final Boss" location
+    loc_final_boss = CCCharlesLocation(player, "Final Boss", None, prism_temple_region)
+    loc_final_boss.place_locked_item(CCCharlesItem("Victory", ItemClassification.progression, None, player))
+    prism_temple_region.locations.append(loc_final_boss)
+
+    # Connect the Regions by named Entrances that must have access Rules
+    menu_region.connect(start_camp_region)
+    menu_region.connect(tony_tiddle_mission_region)
+    menu_region.connect(barn_region)
+    tony_tiddle_mission_region.connect(barn_region, "Barn Door")
```

Comment:
Every explicitly named Entrance that has some rules set in Rules.py connects to a Region that is already connected from the `menu_region` with an Entrance with no rules, so none of the Entrance rules actually do anything.